File: commands/coin.py
#!/usr/bin/env python3
import discord
from random import random


async def coin(ctx):

    embed = discord.Embed(color=0xFEE9B6, title="⏳  Loading...")

    interaction = await ctx.respond(embed=embed)

    coin = "🗣️   Heads!"

    if random() < 0.5:
        coin = "☄️   Tails!"

    embed = discord.Embed(color=0xF6BE3C, title="🪙   Coin Flip", description=coin)

    await interaction.edit_original_response(embed=embed)

File: commands/coin.ts
import {
  DiscordInteraction,
  InteractionResponse,
  Colors,
} from "../types/discord";
import { embedResponse } from "../utils/discord";

export async function handleCoin(
  _interaction: DiscordInteraction,
): Promise<InteractionResponse> {
  const result = Math.random() < 0.5 ? "Heads" : "Tails";
  const emoji = result === "Heads" ? "🪙" : "🔄";

  return embedResponse({
    title: `${emoji} Coin Flip`,
    description: `**${result}!**`,
    color: Colors.GOLD,
  });
}
